core/src/main/java/com/linecorp/armeria/common/logging/RequestLogBuilder.java

```diff
@@ -100,10 +101,15 @@ void session(@Nullable Channel channel, SessionProtocol sessionProtocol, @Nullab
     void serializationFormat(SerializationFormat serializationFormat);
 
     /**
-     * Sets the human-readable name of the {@link Request}, such as RPC method name, annotated service method
-     * name or HTTP method name. This property is often used as a meter tag or distributed trace's span name.
+     * Sets the human-readable service name and method name of the {@link Request} such as:
+     * <ul>
+     *    <li>A service and method name for gRPC and Thrift</li>
+     *    <li>An innermost class and method name for annotated service</li>
+     *    <li>A path pattern and HTTP method name for {@link HttpService}</li>
+     * </ul>
+     * This property is often used as a meter tag or distributed trace's span name.
      */
-    void name(String name);
+    void name(String serviceName, String name);
```

Comment:
I notice `serviceName` is nullable in some places but not here which I guess is the most important. I think serviceName should not be required since not all requests model to having a service name, or users may not be interested (especially for a client, less so for server).

Reply:
Agreed. Could they be separate properties with different availability flag? If no `serviceName()` is set, `defaultServiceName` could be set as a fallback.

Reply:
That was my considering point. Actually, I tried to avoid putting anther tag like `RequestLogProperty.SERVICE_NAMME`.
I think addding `serviceName()` is also a good idea.👍

Reply:
Let me revive the original `name(String name)` and overload `name(serviceName, name)` for  a nullable `serviceName`.